tests/elemental/container_validation.pm

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Test Elemental container image
+#   This image is used as a base to build an Elemental container image.
+#   Then, that image will be used to build a Host OS on top, so
+#   it includes the kernel, firmware, bootloader, etc.
+#
+# Maintainer: elemental@suse.de
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use utils qw(file_content_replace);
+use version_utils qw(is_sle_micro);
+
+=head2 get_filename
+
+ get_filename(file => '/path/to/file');
+
+Extract the filename from F</path/to/file> and return it.
+
+=cut
+
+sub get_filename {
+    my %args = @_;
+
+    my @spl = split('/', $args{file});
+
+    return $spl[$#spl];
+}
+
+sub run {
+    select_serial_terminal;
+
+    my $image = get_required_var('CONTAINER_IMAGE_TO_TEST');
+    my $flavor = get_required_var('FLAVOR');
+    my $arch = get_required_var('ARCH');
+    my $cnt_name = 'elemental_image';
+    my $shared = '/var/shared';
+    my $build_raw_cmd = "elemental --debug build-disk --expandable --squash-no-compression -n $flavor-$arch -o /host --system dir:/";
+
+    # Create shared directory
+    assert_script_run("mkdir -p $shared");
+
+    if ($image =~ m/os-container/) {
+        assert_script_run("podman pull $image");
+        assert_script_run("podman run --name $cnt_name -v $shared:/host:Z -dt $image sleep infinity");
+
+        record_info('Kernel', 'Test that kernel files are present');
+        foreach my $file ('initrd', 'vmlinuz') {
+            validate_script_output("podman exec $cnt_name /bin/sh -c 'ls /boot'", sub { m/$file/ });
+        }
+
+        record_info('Grub', 'Test that /etc/default/grub exists and it is not empty');
```

Comment:
if /etc/default/grub doesn't exist would mean the grub package is missing.

openqa shouldn't test existance of packages (resp. if packages are properly installed) imho.

Reply:
This was an already existing test and I think that it has been added after the issue we had: shipping some Elemental images with a Grub issue. But can be removed later for sure. But I don't have time to do this now (vacations ;-)).